src/main/kotlin/com/coder/gateway/views/CoderGatewayRecentWorkspaceConnectionsView.kt

```diff
@@ -206,94 +199,42 @@ class CoderGatewayRecentWorkspaceConnectionsView(private val setContentCallback:
-                            }
-                                .gap(RightGap.SMALL)
-                            actionButton(
-                                object : DumbAwareAction(
-                                    CoderGatewayBundle.message("gateway.connector.recent-connections.stop.button.tooltip"),
-                                    "",
-                                    CoderIcons.STOP,
-                                ) {
-                                    override fun actionPerformed(e: AnActionEvent) {
-                                        withoutNull(workspaceWithAgent?.workspace, deployment?.client) { workspace, client ->
-                                            jobs[workspace.id]?.cancel()
-                                            jobs[workspace.id] =
-                                                cs.launch(ModalityState.current().asContextElement()) {
-                                                    withContext(Dispatchers.IO) {
-                                                        try {
-                                                            client.stopWorkspace(workspace)
-                                                            fetchWorkspaces()
-                                                        } catch (e: Exception) {
-                                                            logger.error("Could not stop workspace ${workspace.name}", e)
-                                                        }
-                                                    }
-                                                }
-                                        }
-                                    }
-                                },
-                            ).applyToComponent { isEnabled = workspaceWithAgent?.workspace?.latestBuild?.status == WorkspaceStatus.RUNNING }
-                                .gap(RightGap.SMALL)
-                            actionButton(
-                                object : DumbAwareAction(
-                                    CoderGatewayBundle.message("gateway.connector.recent-connections.terminal.button.tooltip"),
-                                    "",
-                                    CoderIcons.OPEN_TERMINAL,
-                                ) {
-                                    override fun actionPerformed(e: AnActionEvent) {
-                                        withoutNull(workspaceWithAgent, deployment?.client) { ws, client ->
-                                            val link = client.url.withPath("/me/${ws.name}/terminal")
-                                            BrowserUtil.browse(link.toString())
-                                        }
-                                    }
-                                },
-                            )
                         }.topGap(gap)
-                        if (deploymentError == null || showError) {
```

Comment:
I think we might want to add this check back or something similar to it.  It is poorly named but basically it was making sure we only display an API error on the first workspace rather than duplicating it on each workspace (it was pretty noisy, especially since the API errors can get long).

For example, say you have 5 workspaces and your token expires, you see a message about being unauthorized and to check your token 5 times on the page.

Or we could group workspaces under a deployment heading and show the API error under the deployment heading, but that might be out of scope for this PR.

Reply:
It looks like atm the moment I actually have this inside the connections.foEach loop (which makes me think with multiple WorkpaceProjectIDE combos it would render multiple time). I didn't notice because I was visually testing with one recent connection.
I'll pull it out of the loop and add the check.

^^^ My adhd is going to quickly become a driving force for building out E2E suites.